gateway/sds_gateway/users/views.py

```diff
@@ -64,6 +73,47 @@
 logger = logging.getLogger(__name__)
 
 
+def is_valid_file(file_path):
+    """Returns True if the path is a valid file for SDS upload."""
+    disallowed_mimes = [
+        "application/x-msdownload",  # .exe
+        "application/x-msdos-program",  # .com
+        "application/x-msi",  # .msi
+    ]
+    mime_type, _ = mimetypes.guess_type(file_path)
+    reasons = []
+    path_obj = Path(file_path)
+    if mime_type in disallowed_mimes:
+        reasons.append(f"Invalid MIME type: {mime_type}")
+    if not path_obj.is_file():
+        reasons.append("Not a file")
+    if path_obj.stat().st_size == 0:
+        reasons.append("Empty file")
+    return len(reasons) == 0, reasons
+
+
+def get_valid_files(root_dir: Path):
+    """
+    Recursively yields (relative_path, file_path) for valid files under root_dir.
+    """
+    for file_path in root_dir.rglob("*"):
+        if not file_path.is_file():
+            continue
+        is_valid, reasons = is_valid_file(str(file_path))
+        if is_valid:
+            rel_path = str(file_path.relative_to(root_dir))
+            yield rel_path, file_path
+
+
+def compute_blake3_checksum(file_path):
+    # Use hashlib.blake2b with digest_size=32 to match 64 hex chars
+    h = hashlib.blake2b(digest_size=32)
+    with file_path.open("rb") as f:
+        for chunk in iter(lambda: f.read(8192), b""):
+            h.update(chunk)
+    return h.hexdigest()
+
+
```

Comment:
these are not called anywhere; let's make them part of a future PR if they'll be used in the future.